synapse/lib/cell.py

```diff
@@ -459,6 +459,43 @@ async def addRole(self, name, iden=None):
     async def delRole(self, iden):
         return await self.cell.delRole(iden)
 
+    async def addUserApiKey(self, name, duration=None, useriden=None):
+        if useriden is None:
+            useriden = self.user.iden
+
+        if self.user.iden == useriden:
+            self.user.confirm(('auth', 'self', 'set', 'apikey'), default=True)
+        else:
+            self.user.confirm(('auth', 'user', 'set', 'apikey'))
+
+        return await self.cell.addUserApiKey(useriden, name, duration=duration)
+
+    async def listUserApiKeys(self, useriden=None):
+        if useriden is None:
+            useriden = self.user.iden
+
+        if self.user.iden == useriden:
+            self.user.confirm(('auth', 'self', 'set', 'apikey'), default=True)
+        else:
+            self.user.confirm(('auth', 'user', 'set', 'apikey'))
+
+        return await self.cell.listUserApiKeys(useriden)
+
+    async def delUserApiKey(self, iden):
+        apikey = await self.cell.getUserApiKey(iden)
+        if apikey is None:
+            mesg = f'User API key with {iden=} does not exist.'
+            raise s_exc.NoSuchObj(mesg=mesg, iden=iden)
```

Comment:
`cell.delUserApiKey` already raises `s_exc.NoSuchIden` if the iden does not exist. We can let that be raised by the cell API and then catch that in the tool.

Reply:
Yeah, but I need to get the api key first so I can auth the user.

Reply:
I changed the exception to NoSuchIden instead.